Description: Remove redundant old-style project declarations.
 Source commits: 13d0d075a91eebf18e16615a732abcaae2b349b0;
  17578cf34877cad67c465335afbf499b88cecce1;
  4cf1f77f2e20b8540aaf8e9478ba6e440dff2a78;
  9618769b8e758f537db180a9389c7ce10c536163;
  2666caef98cd982a09cf76a7c0504c005e38499e;
  2ad54de02c15b48ded23805fb6f399cf9f2a37a2
Origin: upstream
Forwarded: yes
Last-Update: 2026-07-15

diff --git a/Wnn/etc/bdic.c b/Wnn/etc/bdic.c
index 2204dc844af5e5606cfd49d3262da336fb4e5d12..7353beef89fddbcfbb17e52f2967a43454c61280 100644
--- a/Wnn/etc/bdic.c
+++ b/Wnn/etc/bdic.c
@@ -857,7 +857,6 @@ get_kanji_str (UCHAR* kptr, w_char* kanji, w_char* yomi, w_char* comment)
   w_char *tmpy;
   w_char *tmpk;
   w_char *tmpc;
-  extern void Get_kanji_str_r ();
 
   Get_kanji_str_r (kptr, &tmpk, &tmpy, &tmpc);
 
@@ -1306,7 +1305,6 @@ Print_entry (
   char *k;
   char buf[32];
   static w_char revy[LENGTHKANJI];
-  extern void kanji_esc_str ();
 
   if (serial != -1)
     {
diff --git a/Wnn/jserver/daibn_kai.c b/Wnn/jserver/daibn_kai.c
index 5ea6e233122e7877891268843428e1309879c9fe..03598940939e3a1e9252061b6c37fd656b984676 100644
--- a/Wnn/jserver/daibn_kai.c
+++ b/Wnn/jserver/daibn_kai.c
@@ -310,7 +310,6 @@ zentan_able (v, hinsi, fzkchar)
   w_char *fzk_buf;
   int fzkcnt;
   int fzklen;
-  w_char *rev_fzk ();
 #endif /* NO_FZK */
 
   if (hinsi == WNN_ALL_HINSI)
diff --git a/Wnn/jutil/atof.c b/Wnn/jutil/atof.c
index 7e1aba415a93bc9d5a24809988931f536d8d42be..666891b7f6efe6960ca132cde41e04c45de06854 100644
--- a/Wnn/jutil/atof.c
+++ b/Wnn/jutil/atof.c
@@ -289,8 +289,6 @@ static void set_id FRWNN_PARAMS((int, int));
 static void pre_clear_jiritugo_v FRWNN_PARAMS((int));
 static void init FRWNN_PARAMS((int, char **));
 
-extern int wnn_loadhinsi (), create_file_header ();
-
 static void
 error_format (s, d1, d2, d3, d4, d5)
      char *s;
@@ -522,8 +520,6 @@ read_attr ()
     }
 }
 
-extern char *wnn_get_hinsi_name ();
-
 static void
 read_id ()
 {
diff --git a/Wnn/jutil/atorev.c b/Wnn/jutil/atorev.c
index 1e66341a84de1506e87db1cf67557bdbd1699aa6..59370d5e3da1f8bdc1d573ea87d5ec340d1df9c7 100644
--- a/Wnn/jutil/atorev.c
+++ b/Wnn/jutil/atorev.c
@@ -58,15 +58,11 @@
 extern int kanjicount;
 extern struct JT jt;
 
-static void set_rpter1 ();
 static void r_alloc FRWNN_PARAMS((void));
 static void make_ri2 FRWNN_PARAMS((void));
 static void make_ri1 FRWNN_PARAMS((int));
 static void set_rpter1 FRWNN_PARAMS((int));
 
-extern int sort_func_je ();
-extern void sort_kanji ();
-
 struct je **ptmp;
 
 void
diff --git a/Wnn/jutil/dtoa.c b/Wnn/jutil/dtoa.c
index 14529b0061f571c970a11f155d3579223e303edf..93d0d466ed0fc7a2ccd7756e966b20eae2c0f3f4 100644
--- a/Wnn/jutil/dtoa.c
+++ b/Wnn/jutil/dtoa.c
@@ -87,9 +87,6 @@ int input_comment FRWNN_PARAMS((register FILE *));
 int input_hinsi_list FRWNN_PARAMS((register FILE *));
 void input_dic FRWNN_PARAMS((FILE *));
 
-extern void sort (), output_ujis (), exit1 (), init_jeary (), get_kanji_str ();
-extern int input_header (), little_endian (), revdic (), wnn_loadhinsi (), init_heap (), w_stradd (), input_hindo_header (), motoni2 ();
-
 #ifdef CONVERT_with_SiSheng
 int input_sisheng FRWNN_PARAMS((register FILE *ifpter));
 #endif	/* CONVERT_with_SiSheng  */
diff --git a/Wnn/jutil/ujisf.c b/Wnn/jutil/ujisf.c
index e3c4311a9859a550721cc383aa6e3690ad739819..11a3e27964d522d6ec402d1c5dada41bdef2c977 100644
--- a/Wnn/jutil/ujisf.c
+++ b/Wnn/jutil/ujisf.c
@@ -76,7 +76,6 @@ extern unsigned char kanjiaddr (unsigned char *d0, w_char* kanji, w_char* yomi,
 extern void Print_entry (w_char* yomi, w_char* kstr, w_char* cstr,
 		int hindo, int ima, int hinsi, int serial,
 		FILE* ofpter, int esc_exp);
-extern int  w_stradd ();
 
 static void Kanjistradd FRWNN_PARAMS((w_char *, w_char *, w_char *, register UCHAR **));
 static void bunpou_num FRWNN_PARAMS((register char *, register int *));
diff --git a/Wnn/jutil/wnnkill.c b/Wnn/jutil/wnnkill.c
index dde278163177c6dc3d8d3cef8d0c00b7cd9571c9..03c01faaabc9db562b8ad1e5e0913ca1feb63610 100644
--- a/Wnn/jutil/wnnkill.c
+++ b/Wnn/jutil/wnnkill.c
@@ -88,7 +88,6 @@ main (int argc, char** argv)
   static char lang[64] = { 0 };
   char *server_env = NULL;
   char *prog = argv[0];
-  extern char *_wnn_get_machine_of_serv_defs (), *get_server_env ();
 
 /*
   char* p;
@@ -215,13 +214,6 @@ main (int argc, char** argv)
   exit (0);
 }
 
-#ifdef JAPANESE
-extern int eujis_to_jis8 (), eujis_to_sjis ();
-#endif
-#ifdef CHINESE
-extern int ecns_to_big5 ();
-#endif
-
 static void
 out (const char* format, ...)
 {
diff --git a/Wnn/jutil/wnnstat.c b/Wnn/jutil/wnnstat.c
index 71cfd5d01787dec2ce56da0a9dba3c2558de460c..88d549c917eedb60d3707761cd4490297d37cb95 100644
--- a/Wnn/jutil/wnnstat.c
+++ b/Wnn/jutil/wnnstat.c
@@ -80,12 +80,6 @@ struct wnn_ret_buf rb = { 0, NULL };
 extern int optind;
 extern char *optarg;
 
-#ifdef JAPANESE
-extern int eujis_to_jis8 (), eujis_to_sjis ();
-#endif
-#ifdef CHINESE
-extern int ecns_to_big5 ();
-#endif
 static void  err (), usage (), dic (), dic_all (), file_all ();
 static void printall FRWNN_PARAMS((WNN_ENV_INFO *, int));
 static void printpat FRWNN_PARAMS((WNN_ENV_INFO *, int));
@@ -128,7 +122,6 @@ main (argc, argv)
 */
   char *server_env = NULL;
   char *prog = argv[0];
-  extern char *get_server_env ();
 
 /*
     if ((p = getenv("LANG")) != NULL) {
diff --git a/Wnn/jutil/wnntouch.c b/Wnn/jutil/wnntouch.c
index d31cd9defcb7b42c5dc16824918ebf54dd18659e..6182797ace69b8c327c7b6ad0a67c16b1f75d36d 100644
--- a/Wnn/jutil/wnntouch.c
+++ b/Wnn/jutil/wnntouch.c
@@ -54,7 +54,6 @@ static char *rcs_id = "$Id: wnntouch.c,v 1.9 2013/09/02 11:01:39 itisango Exp $"
 
 #include "etc.h"
 
-extern int input_file_header (), check_inode (), change_file_uniq ();
 static void usage (void);
 
 char *com_name;
diff --git a/Wnn/romkan/rk_read.c b/Wnn/romkan/rk_read.c
index b81c553bebc84fa6295cd64e165920e0bec8c348..8d2fba2aa784f9dabe383e6908dc596ab55e4a0e 100644
--- a/Wnn/romkan/rk_read.c
+++ b/Wnn/romkan/rk_read.c
@@ -57,7 +57,6 @@ letter onescan FRWNN_PARAMS((letter**,letter*)),
 char *ename FRWNN_PARAMS((char*));
 
 extern letter *ltrgrow FRWNN_PARAMS((letter*,letter*)),
-  *ltrend FRWNN_PARAMS((letter*)),
   letterpick FRWNN_PARAMS((uns_chr**)),
   *ltr1cut FRWNN_PARAMS((letter*));
 extern char *chrcat FRWNN_PARAMS((char*,char)),
diff --git a/Wnn/uum/sheader.h b/Wnn/uum/sheader.h
index 12e832c072f36f93080ff92a44dbece0cf8c8d2a..58d32732434a8f5bae8f0b12002bed89e36ab1bf 100644
--- a/Wnn/uum/sheader.h
+++ b/Wnn/uum/sheader.h
@@ -244,7 +244,6 @@ extern void clr_line ();
 extern void clr_line_all ();
 extern int connect_jserver (int);
 /* extern int convert_getterm (char *, int);  */ /* moved to include/conv.h  */
-extern int convert_key_setup ();
 extern int cur_ichi (int, int);
 extern void cursor_invisible_raw (void);
 extern void cursor_normal_raw (void);
@@ -460,9 +459,6 @@ extern int call_redraw_line_yincod ();
 extern int hani_settei_yincod ();
 extern void errorkeyin_q ();
 extern int not_call_jl_yomi_len ();
-/* extern int cwnn_pzy_yincod ();  */ /* move to include/etc.h  */
-extern int cwnn_yincod_pzy_str ();
-
 extern int icns_to_ecns ();
 extern int icns_to_big5 ();
 extern int ecns_to_icns ();
diff --git a/Wnn/uum/wnnrc_op.c b/Wnn/uum/wnnrc_op.c
index 9c0135209e12f8bd2d93612800374374df878783..5b405d0d2d1ac2eb8d07f9b6557648820c6fdb6b 100644
--- a/Wnn/uum/wnnrc_op.c
+++ b/Wnn/uum/wnnrc_op.c
@@ -59,10 +59,6 @@
 #include "sheader.h"
 #include "wnn_os.h"
 
-extern int conv_keyin ();
-extern int keyin2 ();
-
-
 void
 getfname (file, path)
      char *path, *file;
